src/middleware/auth.js

```diff
@@ -0,0 +1,54 @@
+'use strict';
+
+'use strict';
+
+const jwt = require('jwt-simple');
+const wrap = require('co-express');
+const createError = require('http-errors');
+
+const EXPIRED = 'Token expired';
+const VALIDATION_FAILED = 'Token validation failed';
+const NOT_CONNECTED = 'User not connected';
+const NEED_KEY = 'IDP secret for JWT is not set';
+const ALGORITHM = 'RS256';
+
+/**
+ * This middleware has been added in order to centralize the code which validates
+ * a jwt token in a single middleware shared by all the micro-services
+ * @param {Boolean} ignoreAuth - Ignore the authentication step
+ * @param {String} jwtKey - The key used to validate a token's signature
+ * @returns {middleware} Express middleware function
+ */
+function setup(ignoreAuth, jwtKey) {
```

Comment:
I suspect that the "ignoreAuth" param is there to bypass authentication for tests and local dev, is it ?
If it's the cas, I strongly disagree with this parameter. I think the right way to go is to provide the right set of fixtures to the developpers to be able to easily test *with authentication*

This is what was done here, and it's very convenient: https://github.com/transcovo/go-chpr-middlewares/tree/master/fixtures